tgui/packages/tgui/interfaces/Manufacturer/blueprintButton.tsx

```diff
@@ -0,0 +1,164 @@
+this just offloads some of the computation to the client.
+
+Consequently, if the checks on the .dm end change, this has to change as well.
+
+Thankfully as this is mostly cosmetic, the only implication is that people might be able
+to try printing blueprints that get refused, or they may not be allowed to fabricate
+blueprints they should be allowed to.
+*/
+const GetProductionSatisfaction = (
+  pattern_requirements:string[],
+  amount_requirements:number[],
+  materials_stored:Resource[]) =>
+{
+  let satisfaction:boolean[] = [];
+  let availableMaterials:Record<string, number> = {};
+  for (let i in pattern_requirements) {
+    let required_pattern = pattern_requirements[i];
+    let compatible_material:Resource;
+    // Try to find compatible resource
+    for (let resource of materials_stored) {
+      if (Object.keys(availableMaterials).find((value:string) => (value === resource.id)) === undefined) {
+        availableMaterials[resource.id] = resource.amount;
+      }
+      if (availableMaterials[resource.id] < amount_requirements[i]/10) {
+        continue;
+      }
+      if (resource.satisfies.find((satisfies_pattern:string) => (
+        (required_pattern === "ALL" || required_pattern === satisfies_pattern)
+      )) === undefined) {
+        continue;
+      }
+      compatible_material = resource;
+      break;
+    }
+    if (compatible_material === undefined) {
+      satisfaction.push(false);
+      continue;
+    }
+    satisfaction.push(true);
+    availableMaterials[compatible_material.id] -= amount_requirements[i];
+  }
+  return satisfaction;
+};
+
+export const BlueprintButton = (props, context) => {
+
+  const { blueprintData, materialData, manufacturerSpeed } = props;
+  const { act } = useBackend(context);
+  const blueprintSatisfaction:boolean[] = GetProductionSatisfaction(
+    blueprintData.item_paths,
+    blueprintData.item_amounts,
+    materialData
+  );
+  const isProduceable = blueprintSatisfaction.find((value:boolean) => !(value)) === undefined;
+  // Don't include this flavor if we only output one item, because if so, then we know what we're making
+  let outputs = (blueprintData.item_outputs.length < 2
+    && blueprintData.create < 2
+    && !blueprintData.apply_material) ? null : (
+      <>
+        <br />
+        Outputs: <br />
+        {blueprintData.item_names.map((value:string, index:number) => (
+          <b key={index}>
+            {blueprintData.create}x {value}<br />
+          </b>
+        ))}
+      </>
+    );
+  let content_requirements = (
+    <Section
+      title="Requirements"
+      buttons={<Button icon="hourglass" backgroundColor="rgba(0,0,0,0)">{getBlueprintTime(blueprintData.time, manufacturerSpeed)}s</Button>}
+    >
+      <LabeledList
+
+      >
+        {blueprintData.material_names.map((value:string, index:number) => (
+          <LabeledList.Item
+            key={index}
+            labelColor={(blueprintSatisfaction[index]) ? "label" : "red"}
```

Comment:
Not sure, but I would imagine the color is fine by default. You also should probably use the semantic(ish) color names, rather than red specifically i.e.:

```suggestion
            labelColor={(blueprintSatisfaction[index]) ? undefined : "bad"}
```